sunpy/coordinates/sun.py

```diff
@@ -716,3 +717,85 @@ def _sun_north_angle_to_z(frame):
         angle = angle[0]
 
     return Angle(angle)
+
+
+def eclipse_amount(observer, *, minimum=False):
+    """
+    Return the percentage of the Sun that is eclipsed by the Moon
+
+    The occultation of the solar disk by the Moon is calculated using the
+    simplifying assumption that the Moon has a constant radius.  Since the Moon has
+    an irregular profile of peaks and valleys, the output can be slightly inaccurate
+    for the start/end of partial solar eclipses (a.k.a. penumbral contacts) and the
+    start/end of total solar eclipses (a.k.a. umbral contacts).
+
+    Parameters
+    ----
+    observer : `~astropy.coordinates.SkyCoord`
+        The observer location and observation time
+    minimum : `bool`
+        Option to use a mean minimum radius for the Moon (R_moon / R_earth =
+        0.272281) rather than the IAU mean radius (R_moon / R_earth = 0.2725076).
+        Defaults to `False`.
+
+    Notes
+    -----
+    The apparent location of the Moon accounts for the effect of light travel time.
+
+    Using the mean minimum radius for the Moon (``minimum=True``) will result in
+    slightly more accurate estimates of the start/end of total solar eclipses, at
+    the expense of slightly more inaccurate estimates for the amount of partial
+    solar eclipses.  See
+    :ref:`this page<https://eclipse.gsfc.nasa.gov/SEmono/reference/radius.html>`__
+    for relevant discussion.
+
+    Examples
+    --------
+    .. minigallery:: sunpy.coordinates.sun.eclipse_amount
+    """
+    # The radius of the Moon to use
+    k = 0.272281 if minimum else 0.2725076
```

Comment:
And also define it better. It's not a radius but a fraction of Earth's radius.

Reply:
It's of course a dimensionless fraction, but it's also the value of the lunar radius in units of Earth radii.

Reply:
I don't know where to stick such constants, since we don't really have a home for non-Sun constants.  I've tacked on a TODO comment, and someone in the future can figure it out.